conans/client/tools/win.py

```diff
@@ -129,7 +129,12 @@ def latest_vs_version_installed(output):
                                  "v80": "8"}
 
 
-def msvs_toolset(settings):
+def msvs_toolset(conanfile):
+    from conans.model.conan_file import ConanFile
+    if isinstance(conanfile, ConanFile):
+        settings = conanfile.settings
+    else:
+        settings = conanfile
```

Comment:
Ok, I see the usage here: https://github.com/conan-io/conan-center-index/blob/d636ba2fa342b74ac91e8277c4eab4ce23dc7257/recipes/jemalloc/all/conanfile.py

Reply:
But this will not be broken with this, right?